src/Raven.Server/Rachis/Follower.cs

```diff
@@ -672,36 +672,76 @@ private unsafe bool ReadSnapshot(SnapshotReader reader, ClusterOperationContext
                         return true;
                     case RootObjectType.VariableSizeTree:
                         size = reader.ReadInt32();
-                        reader.ReadExactly(size);
+                        reader.ReadExactly(size); 
 
                         Tree tree = null;
+                        Slice.From(context.Allocator, reader.Buffer, 0, size, ByteStringType.Immutable, out Slice treeName); // The Slice will be freed on context close
+
+                        entries = reader.ReadInt64();
+                        var flags = TreeFlags.FixedSizeTrees;
+
                         if (dryRun == false)
                         {
-                            Slice.From(context.Allocator, reader.Buffer, 0, size, ByteStringType.Immutable, out Slice treeName); // The Slice will be freed on context close
                             txw.DeleteTree(treeName);
                             tree = txw.CreateTree(treeName);
                         }
 
-                        entries = reader.ReadInt64();
+                        if (_connection.Features.MultiTree)
+                            flags = (TreeFlags)reader.ReadInt32();
+
                         for (long i = 0; i < entries; i++)
                         {
                             token.ThrowIfCancellationRequested();
+                            // read key
                             size = reader.ReadInt32();
                             reader.ReadExactly(size);
+
                             using (Slice.From(context.Allocator, reader.Buffer, 0, size, ByteStringType.Immutable, out Slice valKey))
                             {
-                                size = reader.ReadInt32();
-                                reader.ReadExactly(size);
-
-                                if (dryRun == false)
+                                switch (flags)
                                 {
-                                    using (tree.DirectAdd(valKey, size, out byte* ptr))
-                                    {
-                                        fixed (byte* pBuffer = reader.Buffer)
+                                    case TreeFlags.None:
+                                    case TreeFlags.FixedSizeTrees:
```

Comment:
If we have `TreeFlags.FixedSizeTrees` then shouldn't we add the received items to `FixedSizeTree`? As I see we only have `Tree tree = null;` instance

Reply:
We don't send any `FixedSizeTrees` over the snapshot. `case TreeFlags.FixedSizeTrees:` was removed